hw/ip/otbn/dv/otbnsim/sim/dmem.py

```diff
@@ -61,7 +61,15 @@ def __init__(self) -> None:
         self.trace = []  # type: List[Trace]
 
     def _get_u32s(self, idx: int) -> List[int]:
-        '''Return the value at idx as 8 uint32's in little-endian order'''
+        '''Return the value at idx as 8 little-endian uint32's
+
+        These are ordered by increasing address, so the first word from
+        _get_u32s(0) will correspond to bytes at addresses 0x0 through 0x3.
+        Since the 32-bit unsigned ints are themselves little-endian, the byte
+        at address 0x0 (which holds the LSB for the first 256-bit value) will
+        be the LSB of the first u32.
```

Comment:
The first sentence sounds good. The second sentence confuses me and I'd just remove it. An int is a number/value, not a sequence of bytes. As such, endianness doesn't matter. We want to do arithmetic with the ints returned by this function, assuming they are numbers that Python can work with.

Reply:
Hmm. This is the DMEM class, so we're passing around stuff that maps to specific addresses in memory. I wrote this sentence when trying to get all the endian-ness right: it took several goes and various statements saying "Hey, Rupert of the future: Rupert of the past thought things should be little-endian here" or similar turned out to be rather helpful.

Maybe it would make more sense changed to something like this? "Since 32-bit loads from memory are also little-endian, the byte address 0x0 ..." ?

Reply:
Let's do that.